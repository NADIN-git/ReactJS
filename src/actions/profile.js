export const CHANGE_NAME = 'PROFILE::CHANGE_NAME'
export const CHANGE_IS_ONLINE = 'PROFILE::CHANGE_IS_ONLINE'

export const changeName = (surname,name,patronymic) => ({
    type: CHANGE_NAME,
    payload: {
        surname,
        name,
        patronymic,
    },    
})

export const changeViewCheckbox = (viewCheckbox) => ({   
    type: CHANGE_IS_ONLINE,
    payload: {
        viewCheckbox,
    },
})
