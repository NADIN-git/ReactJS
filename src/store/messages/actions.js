//import { AUTHORS } from '../../components/App/Constants'
export const ADD_MESSAGE = 'MESSAGES::ADD_MESSAGE';

export const addMessage = (chatId, message) => ({
  type: ADD_MESSAGE,
  payload: {
    chatId,
    message,
  },
})

export const addMessageWithThunk = (chatId, message) => {
  return (dispatch, getState) => {
    console.log(getState())

    setTimeout(() => {
      dispatch(addMessage(chatId, message))
    }, 1000)
  }
}