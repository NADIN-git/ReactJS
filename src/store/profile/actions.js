import firebase from 'firebase'
export const CHANGE_NAME = 'PROFILE::CHANGE_NAME'
export const CHANGE_IS_ONLINE = 'PROFILE::CHANGE_IS_ONLINE'
export const CHANGE_IS_AUTHED = 'PROFILE::CHANGE_IS_AUTHED'


export const changeName = (surname,name,patronymic) => ({
    type: CHANGE_NAME,
    payload: {
        surname: "Колотилова",
        name: "Надежда",
        patronymic: "Викторовна",
    },    
})

export const changeIsAuthed = (isAuthed) => ({
    type: CHANGE_IS_AUTHED,
    payload: {
        isAuthed,
    },
})

export const addProfilToDatabase = (surname, name, patronymic) => {
    return () => {
        firebase.database().ref('profile').push({ surname, name, patronymic })
    }
}

export const subscribeOnProfilChangings = (surname, name, patronymic) => {
    return (dispatch, getState) => {
        firebase
            .database()
            .ref('profile')
            .child(surname, name, patronymic)
            .on('child_added', (snapshot) => {
                console.log('child_added', snapshot.val())
  
                dispatch(changeName(surname, name, patronymic, snapshot.val()))
            })
  
        firebase
            .database()
            .ref('profile')
            .child(surname, name, patronymic)
            .on('child_changed', (snapshot) => {
                console.log('child_changed', snapshot.val())
  
                dispatch(surname, name, patronymic(surname, name, patronymic, snapshot.val()))
            })
    }
  }

export const changeNameWithThunk = (surname,name,patronymic) => {
    return (dispatch, getState) => {
        console.log(getState())

        setTimeout(() => {
            dispatch(changeName(surname,name,patronymic))
        }, 1000)
    }
}


export const changeViewCheckbox = (viewCheckbox) => ({   
    type: CHANGE_IS_ONLINE,
    payload: {
        viewCheckbox,
    },
})

export const changeViewCheckboxWithThunk = (viewCheckbox) => {
    return (dispatch, getState) => {
        console.log(getState())

        setTimeout(() => {
            dispatch(changeViewCheckbox(viewCheckbox))
        }, 1000)
    }
}
