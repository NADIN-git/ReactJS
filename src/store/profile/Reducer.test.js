import { render, screen } from "@testing-library/react";
import Reducer from "./Reducer";
import { changeIsAuthed, changeName, changeViewCheckbox } from './actions'

describe("Reducer", () => {
  const state = {
    surname: "",
    name: "",
    patronymic: "",
    viewCheckbox: false,
    isAuthed: false,
  }
  it('Change isAuthed', () => {
    const action = changeIsAuthed(true)
    const newStateIsAuthed = Reducer(state, action)
    expect(newStateIsAuthed.isAuthed).toBe(true)
    expect(newStateIsAuthed).toMatchSnapshot()
  })
  it('Change viewCheckbox', () => {
    const action = changeViewCheckbox(true)
    const newStateCheckbox = Reducer(state, action)
    expect(newStateCheckbox.viewCheckbox).toBe(true)
    expect(newStateCheckbox).toMatchSnapshot()
  })
  it('Change surname, name, patronymic', () => {
    const action = changeName("Колотилова", "Надежда", "Викторовна")
    const newStateName = Reducer(state, action)
    expect(newStateName.surname).toBe('Колотилова')
    expect(newStateName.name).toBe("Надежда")
    expect(newStateName.patronymic).toBe("Викторовна")
    expect(newStateName).toMatchSnapshot()
  })  
})