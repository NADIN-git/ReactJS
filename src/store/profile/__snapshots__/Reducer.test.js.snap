// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer Change isAuthed 1`] = `
Object {
  "isAuthed": true,
  "name": "",
  "patronymic": "",
  "surname": "",
  "viewCheckbox": false,
}
`;

exports[`Reducer Change surname, name, patronymic 1`] = `
Object {
  "isAuthed": false,
  "name": "Надежда",
  "patronymic": "Викторовна",
  "surname": "Колотилова",
  "viewCheckbox": false,
}
`;

exports[`Reducer Change viewCheckbox 1`] = `
Object {
  "isAuthed": false,
  "name": "",
  "patronymic": "",
  "surname": "",
  "viewCheckbox": true,
}
`;
